spot-client/src/common/backend/utils.js

```diff
@@ -0,0 +1,146 @@
+import { logger } from 'common/logger';
+import { getDeviceId } from 'common/utils/device-id';
+
+/**
+ * FIXME duplicated with post-to-endpoint and the jitter calculation in the loaders.
+ *
+ * Gets next timeout using the full jitter pattern.
+ *
+ * @param {number} retry - The retry number. It's 1 on the first retry.
+ * @returns {number} - The amount of waiting before trying another time given in milliseconds.
+ * @private
+ */
+function _getNextTimeout(retry) {
+    // 1st retry 0 - 2 seconds
+    // 2nd retry 0 - 4 seconds
+    // 3rd retry 0 - 16 seconds
+    return Math.floor(Math.random() * Math.pow(2, retry) * 1000);
+}
+
+/**
+ * Sends HTTP request using {@code fetch} with retries on network and server errors.
+ *
+ * @param {Object} fetchOptions - Options related to the fetch request.
+ * @param {string} fetchOptions.url - The URL to be passed to the fetch function.
+ * @param {Object} fetchOptions.requestOptions - The options to be passed to the fetch function.
+ * @param {string} fetchOptions.operationName - The name of the fetch operation that will appear in
+ * the retry related log entries.
+ * @param {number} retry - Keeps track of the retry attempt.
+ * @param {number} maxRetries - How many times will retry the request.
+ * @returns {Promise<Object>} - A promise resolved with the JSON parsed from the response.
+ */
+function fetchWithRetry(fetchOptions, retry = 0, maxRetries = 3) {
```

Comment:
I think we can try to do something fancy and create a new function within the scope to avoid having retry and max retry as parameters to the signature. It's not a big deal but maybe later we can try it and maybe you'll love it.

Reply:
I don't know how to do it. How would I remove the `retry` argument ?

Reply:
We make a new function within this function that wraps all the logic and either checks retry/maxRetries in its scope or accepts it in the signature. I can give it a shot in an exploratory follow-up PR.

Reply:
okay, sounds good